nni/retiarii/experiment/pytorch.py

```diff
@@ -81,7 +81,7 @@ def __setattr__(self, key, value):
         if key in fixed_attrs and fixed_attrs[key] != value:
             raise AttributeError(f'{key} is not supposed to be set in Retiarii mode by users!')
         # 'trial_code_directory' is handled differently because the path will be converted to absolute path by us
-        if key == 'trial_code_directory' and not (value == Path('.') or os.path.isabs(value)):
+        if key == 'trial_code_directory' and not (str(value) == '.' or os.path.isabs(value)):
```

Comment:
what is the difference?

Reply:
In init, it's a string. So the check failed.